cpp/include/cuvs/neighbors/detail/ivf_flat_search.cuh

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
```

Comment:
These files should all be moved to src/. The only files in include/ should be public API headers that we would like the users to see.